src/main/java/com/mixpanel/android/util/ImageStore.java

```diff
@@ -93,7 +93,18 @@ public Bitmap getImage(String url) throws CantGetImageException {
                 throw new CantGetImageException("Downloaded data could not be interpreted as a bitmap");
             }
         } else {
-            bitmap = BitmapFactory.decodeFile(file.getAbsolutePath());
+            BitmapFactory.Options option = new BitmapFactory.Options();
+            option.inJustDecodeBounds = true;
+            BitmapFactory.decodeFile(file.getAbsolutePath(), option);
+            Long imageSize = new Long(option.outHeight * option.outWidth);
+            Long freeMemory = Runtime.getRuntime().freeMemory();
+            if (imageSize > freeMemory) {
+                throw new CantGetImageException("Do not have enough memory for the image");
+            }
+
+            option.inJustDecodeBounds = false;
+            bitmap = BitmapFactory.decodeFile(file.getAbsolutePath(), option);
```

Comment:
I wrote the comment on line 106. Re-read my comment :) We do not need to change the flag and pass `option`.